R/compile_report.R

```diff
@@ -231,15 +234,46 @@ compile_report <- function(file, quiet = FALSE, factory = getwd(),
   }
 
   dir.create(output_dir, FALSE, TRUE)
-
+  
+  output_files <- list()
   for (file in new_files) {
     destination <- file.path(output_dir, file)
     move_file(file, destination)
+    i <- match(file, new_files) #gets index
+    output_files[[i]] <- destination
   }
-
+  
   ## remove empty directories
   to_remove <- file.path(new_dirs)
   file.remove(to_remove)
+  
+  
+  # ================
+  ### ! KEEP AT END OF FUNCTION ! ###
+  ## Logging approach: 
+  ## 1. Combine all env variables and args into a list and add timestamp
+  ## 2. Read exising log file into a dataframe
+  ## 3. Add row to log dataframe and write new csv
+  
+  
+  ## Start log code
+  setwd(odir)
+  if (sum(file.exists(".compile_log.rds", hidden.files = TRUE)) == 0) {
+    initialize_log <- list(initialize = TRUE, timestamp = Sys.time())
+    saveRDS(initialize_log, ".compile_log.rds")
+  }
+  log_entry$timestamp <- Sys.time()
```

Comment:
The timestamp should probably be the same for both the output files (used to name the `compiled_[params]_[timestamp]/` folder) and the log. Store it upstream and use here?